cleanlab/object_detection/summary.py

```diff
@@ -135,27 +135,29 @@ def visualize(
             _, _ = _draw_boxes(
                 fig, axes[1], pbbox, plabels, edgecolor="b", linestyle="-.", linewidth=1
             )
-
+    bbox_extra_artists = None
     if label or prediction is not None:
         legend, plt = _plot_legend(class_names, label, prediction)
-        if save_path:  # save with legend
-            plt.savefig(
-                save_path,
-                format="pdf",
-                bbox_extra_artists=(legend,),
-                bbox_inches="tight",
-                transparent=True,
-                pad_inches=0.5,
-            )
-    elif save_path:
+        bbox_extra_artists = (legend,)
+
+    if save_path:
+        allowed_image_formats = set(["png", "pdf", "ps", "eps", "svg"])
+        image_format = (
+            save_path[-3:]
+            if len(save_path) > 2
+            and save_path[-4] == "."
```

Comment:
I believe this check will fail for `filename.ps` since `save_path[-4] != "."` 
Would it instead be better to use `save_path.split(".")`? Tho you might need some try-except blocks to catch in case the path provided does not contain a period